src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -688,7 +690,11 @@ DistributedInsertSelectSupported(Query *queryTree, RangeTblEntry *insertRte,
 	}
 
 	/* we don't support LIMIT, OFFSET and WINDOW functions */
-	DeferredErrorMessage *error = MultiTaskRouterSelectQuerySupported(subquery);
+    bool requireSubqueryPushdownCondsForTopLevel = true;
```

Comment:
While using limit / offset clauses in top-level query doesn't prevent a query from being handled by router planner, this is not the case for distributed insert select. This is because, a "distributed insert select" doesn't necessarily mean targeting a single worker node.